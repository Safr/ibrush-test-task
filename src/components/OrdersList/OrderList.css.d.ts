export const List: string;
export const list: string;
