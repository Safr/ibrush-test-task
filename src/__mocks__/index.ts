export const orderBook = {
  BTC_USD: {
    ask: ['6891', '0.005', '34.455'],
    ask_amount: '2413543.16939829',
    ask_quantity: '190.96765367',
    ask_top: '6887.20157116',
    bid: ['6849.7394927', '0.75', '5137.30461952'],
    bid_amount: '286086.48310063',
    bid_quantity: '11524.76654836',
    bid_top: '6875',
  },
};

export const order = {
  avg: '6875.86448007',
  buy_price: '6859.227881',
  high: '6969.54201841',
  last_trade: '6864.01',
  low: '6800',
  sell_price: '6873.71642794',
  updated: 1528752429,
  vol: '1112.01492053',
  vol_curr: '7632881.53471028',
};

export const currentOrder = {
  BTC_USD: {
    avg: '6875.86448007',
    buy_price: '6859.227881',
    high: '6969.54201841',
    last_trade: '6864.01',
    low: '6800',
    sell_price: '6873.71642794',
    updated: 1528752429,
    vol: '1112.01492053',
    vol_curr: '7632881.53471028',
  },
};

export const orders = { currentOrder };
export const changedOrders = [''];
export const orderBookInfo = [['5897.8242', '0.003', '17.6934726']];

// export const fetchOrders = () => Promise.resolve({
//   orders: {
//     BTC_USD: {
//       avg: '6875.86448007',
//       buy_price: '6859.227881',
//       high: '6969.54201841',
//       last_trade: '6864.01',
//       low: '6800',
//       sell_price: '6873.71642794',
//       updated: 1528752429,
//       vol: '1112.01492053',
//       vol_curr: '7632881.53471028',
//     },
//   },
// });
