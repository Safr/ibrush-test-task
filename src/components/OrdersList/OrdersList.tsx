import React from 'react';
import cuid from 'cuid';
import classes from './OrderList.css';
import OrderItem from '../OrderItem/OrderItem';
import { OrderInterface } from '../../types';

interface Props {
  orders: object;
  changedOrders: string[];
  onOpenPopup: (orderName: string, className: string, order: OrderInterface) => void;
}

const OrdersList: React.SFC<Props> = ({ orders, changedOrders, onOpenPopup }) => {
  return (
    <ul className={classes.List}>
      {Object.keys(orders).slice(0, 10).map((orderName: string, idx: number) => (
        <OrderItem
          key={orderName}
          orderName={orderName}
          order={orders[orderName]}
          changedPrice={changedOrders !== null ? changedOrders[idx] : ''}
          onOpenPopup={onOpenPopup}
        />
      ))}
    </ul>
  );
};

export default OrdersList;
