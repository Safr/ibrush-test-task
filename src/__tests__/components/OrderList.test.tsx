import React from 'react';
import { shallow } from 'enzyme';
import OrdersList from '../../components/OrdersList/OrdersList';
import { orders, changedOrders } from '../../__mocks__';

describe('OrdersList Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(
      // tslint:disable-next-line:jsx-no-lambda
      <OrdersList orders={orders} changedOrders={changedOrders} onOpenPopup={() => null} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
