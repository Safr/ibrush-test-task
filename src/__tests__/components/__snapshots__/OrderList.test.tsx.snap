// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersList Component should render correctly 1`] = `
<ul>
  <OrderItem
    changedPrice=""
    key="currentOrder"
    onOpenPopup={[Function]}
    order={
      Object {
        "BTC_USD": Object {
          "avg": "6875.86448007",
          "buy_price": "6859.227881",
          "high": "6969.54201841",
          "last_trade": "6864.01",
          "low": "6800",
          "sell_price": "6873.71642794",
          "updated": 1528752429,
          "vol": "1112.01492053",
          "vol_curr": "7632881.53471028",
        },
      }
    }
    orderName="currentOrder"
  />
</ul>
`;
